server/sonar-db-dao/src/main/java/org/sonar/db/metric/MetricDao.java

```diff
@@ -81,7 +82,7 @@ public int countEnabled(DbSession session, @Nullable Boolean isCustom) {
   }
 
   public MetricDto insert(DbSession session, MetricDto dto) {
-    mapper(session).insert(dto);
+    mapper(session).insert(validate(dto));
```

Comment:
AFAIK, It's not the responsability of DAO to validate dto. This should be in a higher layer, a XXXUpdater or the WS